src/bootstrap/bin/rustc.rs

```diff
@@ -358,7 +389,9 @@ fn main() {
         }
     }
 
-    let code = exec_cmd(&mut cmd).unwrap_or_else(|_| panic!("\n\n failed to run {:?}", cmd));
+    let code = exec_cmd(&mut cmd).unwrap_or_else(|_| {
+        panic!("\n\n failed to run {:?}\n\n with env: {}", cmd, env.to_string())
```

Comment:
if you put it before the command, one can copy past the env vars and the command in one go